### PR TITLE
Add More Sentry

### PR DESCRIPTION
**Pull request by richarddubay**
Added more Sentry logging for more/better information on problems our people are seeing.

## Creator

- [x] Build relevant tests if any apply
- [x] Ensure there are no new warnings
- [x] Upload an animated GIF showcasing the solution (if it applies)
- [x] Set a relevant reviewer

## Reviewer

- [ ] Run `test` and make sure all tests pass
- [ ] Review function and form on iOS
- [ ] Review function and form on Android
- [ ] Review function and form on Web
- [ ] Review new test logic

### CODE REVIEW

File: src/@utils/removeUndefined.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
 // module({ thing: undefined, data: 0 }) => { data: 0 }
+import sentry from '@utils/sentry';
+
 export default function (obj) {
-  return JSON.parse(JSON.stringify(obj));
+  let returnValue = {};
+  try {
+    returnValue = JSON.parse(JSON.stringify(obj));
+  } catch (e) {
+    sentry.captureException(e);
```
[the remark this answers is not in the document]
Comment (richarddubay) [reply src/@utils/removeUndefined.js]: So the `removeUndefined` is used in `createValidationOrder` and `getOrderDetails`. I would agree that at least the `Invalid JSON` error we're already seeing _probably_ isn't being caused by this. I can remove that one if you think it's unnecessary.

File: src/@data/withGive/resolvers/mutations.js [hunk 2]
```diff
@@ -374,12 +363,18 @@ export async function setSavedPaymentMethod(result, variables, { cache }) {
 }
 
 export function restoreContributions(result, variables, { cache }) {
+  let stateVariables = {};
+  try {
+    stateVariables = JSON.parse(variables.state);
+  } catch (e) {
+    sentry.captureException(e);
```
[the remark this answers is not in the document]
Comment (richarddubay) [reply src/@data/withGive/resolvers/mutations.js]: I believe this ends up being used in or near the `restoredCheckout` ... which is where the majority of the `Invalid JSON` errors seem to be coming from. So this feels different than the `removeUndefined` one. I'll see what I can do about possibly logging other information here and not _just_ the error.

**Comment by richarddubay** [comment]
@conrad-vanl I was able to pull a little more data and actually bring back the `result` and the `variables`. It'll be interesting to see what we get with this.

File: src/@data/withGive/resolvers/mutations.js [hunk 3]
```diff
@@ -374,12 +363,23 @@ export async function setSavedPaymentMethod(result, variables, { cache }) {
...
+        result,
+        variables,
+      },
+    });
+  }
+
   cache.writeQuery({
     query: contributionsQuery,
     data: {
-      contributions: JSON.parse(variables.state),
```
Comment (conrad-vanl) [inline src/@data/withGive/resolvers/mutations.js]: will this contain the user's creditcard / bankaccount info?

Comment (richarddubay) [reply src/@data/withGive/resolvers/mutations.js]: That's a great question ... that I don't know the answer to. I didn't look at what was in them, I just wanted to log them. I'll add some console logs and do a bit more looking and get back to you.

Comment (richarddubay) [reply src/@data/withGive/resolvers/mutations.js]: @conrad-vanl This should be all set now. Thanks for taking a look at it!

**Pull request merged**